packages/docs/src/styleguide.md

```diff
@@ -33,9 +33,41 @@ relevant sections in the sidebar for details about the styling & component APIs
   Most packages consist of JS and CSS resources, so please make sure you're including both in your application.
 </div>
 
+### TypeScript
+
+Blueprint is written in [TypeScript](https://www.typescriptlang.org/), a typed superset of
+JavaScript that compiles to plain JavaScript. Many of the code samples throughout this site and all
+interactive examples are also written in TypeScript. TypeScript code looks exactly like ES2015 code
+(because it literally is) with the addition of type signatures.
+
+```ts
+// variables
+const variableName: varType;
+const name: string;
+const disabled: boolean;
+
+// functions (and function variables)
+function funcName(arg1: argType, arg2: argType): returnType { }
+const funcName: (arg1: argType) => returnType;
+function split(str: string, delim: string): string[] { }
+function map<T, U>(array: T[], iterator: (item: T, index: number) => U): U[];
+
+// interfaces describe plain objects
+// (we use the convention that interfaces begin with "I")
+interface IOption {
+  label: string;
+  value: string;
+}
+const option: IOption = { label: "Name", value: "gilad" };
+```
+
+**You do not need to use TypeScript to consume Blueprint** (but major "props" if you do). Familiarity
```

Comment:
it's usually up to the browser/OS to display emojis... but I also know there are web fonts to display them. I'm not kidding but obvs low priority